Maps trail line

jurintal1 opened the pull request:
Drawing the trail based on trail node coordinates and map height in those locations.

Namirual commented:
The line rendering works, however:

1) TrailDisplayTest is broken by the way the lines are generated in script, because making changes to renderers in tests does not seem to be allowed. A work-around is needed.

2) The TrailGenerator change should be reverted, there's just a needless change in indentation.

jurintal1 commented:
1. Changed this to use sharedMaterial instead of material, which avoided the error. Also changed the tests to acknowledge the increased number of children to TrailDisplay.
2. TrailGenerator returned to previous form.
